docs/admin/kubeadm-init.md

```diff
@@ -0,0 +1,387 @@
+In summary, `kubeadm init --feature-gates=SelfHosting=true` works as follows:
+
+  1. As usual, kubeadm creates static pod YAML files in `/etc/kubernetes/manifests/`.
+
+  1. Kubelet loads these files and launches the initial static control plane.
+    Kubeadm waits for this initial static control plane to be running and
+    healthy. This is identical to the `kubeadm init` process without self-hosting.
+
+  1. Kubeadm uses the static control plane pod manifests to construct a set of
+    DaemonSet manifests that will run the self-hosted control plane.
+
+  1. Kubeadm creates DaemonSets in the `kube-system` namespace and waits for the
+    resulting pods to be running.
+
+  1. Once the new control plane is running (but not yet active), kubeadm deletes
+    the static pod YAML files. This triggers kubelet to stop those static pods.
+
+  1. When the original static control plane stops, the new self-hosted control
+    plane is able to bind to listening ports and become active.
+
+This process (steps 3-6) can also be triggered with `kubeadm phase selfhosting convert-from-staticpods`.
+
+### Running kubeadm without an internet connection
+
+All of the control plane components run in Pods started by the kubelet and
+the following images are required for the cluster works will be automatically
+pulled by the kubelet if they don't exist locally while `kubeadm init` is initializing
+your master:
+
+| Image Name |  v1.7 release branch version | v1.8 release branch version
+|---|---|---|
+| gcr.io/google_containers/kube-apiserver-${ARCH} | v1.7.x | v1.8.x
+| gcr.io/google_containers/kube-controller-manager-${ARCH} | v1.7.x | v1.8.x
+| gcr.io/google_containers/kube-scheduler-${ARCH} | v1.7.x | v1.8.x
+| gcr.io/google_containers/kube-proxy-${ARCH} | v1.7.x | v1.8.x
+| gcr.io/google_containers/etcd-${ARCH} | 3.0.17 | 3.0.17
+| gcr.io/google_containers/pause-${ARCH} | 3.0 | 3.0
+| gcr.io/google_containers/k8s-dns-sidecar-${ARCH} | 1.14.4 | 1.14.4
+| gcr.io/google_containers/k8s-dns-kube-dns-${ARCH} | 1.14.4 | 1.14.4
+| gcr.io/google_containers/k8s-dns-dnsmasq-nanny-${ARCH} | 1.14.4 | 1.14.4
+
+Here `v1.7.x` means the "latest patch release of the v1.7 branch".
+
+`${ARCH}` can be one of: `amd64`, `arm`, `arm64`, `ppc64le` or `s390x`.
+
+### Automating kubeadm
+
+Rather than copying the token you obtained from `kubeadm init` to each node, as
+in the [basic kubeadm tutorial](/docs/admin/kubeadm/), you can parallelize the
+token distribution for easier automation. To implement this automation, you must
+know the IP address that the master will have after it is started.
+
+1.  Generate a token. This token must have the form  `<6 character string>.<16
+    character string>`.  More formally, it must match the regex:
+    `[a-z0-9]{6}\.[a-z0-9]{16}`.
+
+    kubeadm can generate a token for you:
+
+    ```bash
+    kubeadm token generate
+    ```
+
+1. Start both the master node and the worker nodes concurrently with this token.
```

Comment:
Doesn't the worker node need to start up _after_ the master's apiserver is running?